server/src/test/java/org/elasticsearch/common/settings/ScopedSettingsTests.java

```diff
@@ -1221,4 +1256,54 @@ public String getKey(final String key) {
                 equalTo(oldSetting.get(settings).stream().map(s -> "new." + s).collect(Collectors.toList())));
     }
 
+    public void testUpdateOfValidationDependantSettings() {
+        ClusterSettings service = new ClusterSettings(Settings.EMPTY, new HashSet<>(Arrays.asList(SETTING_FOO_LOW, SETTING_FOO_HIGH)));
+
+        Settings.Builder target = Settings.builder();
+        Settings.Builder updates;
+        boolean updated;
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertTrue(updated);
+        assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+        assertThat(updates.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertFalse(updated);
```

Comment:
I think it's ok to return `true` here. Claiming to have updated a setting when really it hasn't changed in value is a right-side failure: it means we call `reroute()` to reallocate any shards affected by the new settings, but this does nothing if the setting change was a no-op. On the other hand failing to call `reroute()` when a setting _does_ change is very bad, so changing the behaviour here needs a lot more scrutiny.

Reply:
I understand. I'll restore the previous behavior to not break the current data flow and leave this responsibility to downstream components, it's safer.

I wasn't very at ease with this change, thank you for catching it. I'll dig deeper into it with a debugger just for my own curiosity.